### PR TITLE
perf(weighted_median): use quickselect to calculate the static median

### PR DESCRIPTION
**Pull request by jrs65**
This moves to using a weighted quickselect algorithm for calculating the
*static* weighted median which is significantly faster (> 50x) than the
original AVL tree code.

### CODE REVIEW

**Comment by jrs65** [comment]
@nritsche any hint with the readthedocs error?

**Comment by nritsche** [comment]
Oh, I must have activated docs builds for this, too. But it won't work. We can ignore it though, it's not used for branch protection yet.

I started reviewing (and github made a mess out of the diff). What was in your force push?

**Comment by jrs65** [comment]
Nothing really. I had enabled a pytest plugin that will run doctests for Cython code, but it doesn't work with the newest Pytest and so I ended up disabling it for now.

File: caput/weighted_median.pyx [hunk 1]
```diff
@@ -9,99 +9,366 @@ from MedianTree cimport Tree, Data
...
+        raise TypeError("Type of data array is not supported.")
 
-    if data.ndim is 1:
-        return _weighted_median_1D(data, weights, c_method)
-    if data.ndim is 2:
-        return _weighted_median_2D(data, weights, c_method)
-    raise NotImplementedError('weighted_median is only implemented for 1 and 2 dimensions, not {}'
-                              .format(data.ndim))
+    if A.ndim == 1:
+        return res[0]
```
Comment (nritsche) [inline caput/weighted_median.pyx:313]: I don't understand what this case is about.

**Review by nritsche — APPROVED** [review]
This looks pretty perfect.

My review is mostly questions and comments I added after I understood what something does.

File: caput/weighted_median.pyx (hunk 1 above)
Comment (jrs65) [reply caput/weighted_median.pyx:313]: It's so that the routine returns a scalar if the input array was only 1D.

Comment (nritsche) [reply caput/weighted_median.pyx:313]: Oh yeah, it does make sense :+1:

**Comment by jrs65** [comment]
> Oh, I must have activated docs builds for this, too. But it won't work. We can ignore it though, it's not used for branch protection yet.

I've deactivated the readthedocs check for the time being, so hopefully we'll get rid of those red X's

**Pull request merged**